pkg/udn/primary_network.go

```diff
@@ -0,0 +1,37 @@
+package udn
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubevirt/ipam-extensions/pkg/config"
+)
+
+func FindPrimaryNetwork(ctx context.Context,
```

Comment:
Please consider passing the nadList as an input parameter to this function. 
It would make unit-testing this function easier to do.

Reply:
This can be done at not public function so we do unit test, Public intgerface should not pass nadList

Reply:
What I mean is, Handle(webhook)/Reconcile(Controller) can list the nads, mapping them once:
```
map[string]*NAD{
   'ns/nadName1`: *NAD
   'ns/nadName2`: *NAD
   `primary`: *NAD
```
then just send the map to each function case (secondary, primary)..
But only if it works for you. If it doesn't add up them I can try it on a follow up PR..